R/utilities-mass-balance.R

```diff
@@ -8,277 +8,244 @@
-
-  # Get all the relevant compounds
-  organism <- ospsuite::getContainer("Organism", simulation)
-  allCompoundNames <- c(simulation$allFloatingMoleculeNames(), simulation$allStationaryMoleculeNames())
-  relevantCompoundNames <- allCompoundNames[!sapply(allCompoundNames, function(compoundName) {
-    compoundName %in% simulation$allEndogenousStationaryMoleculeNames()
-  })]
-
-  # User defined compound selection
-  selectedCompoundNames <- settings$selectedCompoundNames %||% relevantCompoundNames
-  validateIsIncluded(selectedCompoundNames, relevantCompoundNames)
-
-  # Get all the molecule paths (with dimension=amount) of the selected/relevant coumpounds
-  molecules <- ospsuite::getAllMoleculesMatching(paste0("**|", selectedCompoundNames), organism)
-
-  # Clear concentration output in case any concentrations are still included
+  allMoleculePaths <- ospsuite::getAllMoleculePathsIn(simulation)
   ospsuite::clearOutputs(simulation)
-  for (molecule in molecules) {
-    addOutputs(quantitiesOrPaths = molecule, simulation = simulation)
-  }
-
+  addOutputs(quantitiesOrPaths = allMoleculePaths, simulation = simulation)
   simulationResults <- ospsuite::runSimulation(simulation)
-  simulationResultsOutput <- ospsuite::getOutputValues(
-    simulationResults = simulationResults,
-    quantitiesOrPaths = simulationResults$allQuantityPaths
-  )
 
-  # Create a data.frame with full string path and separates the last 3 elements of paths
-  # to perform filtering and grouping of for the final plot
-  pathsArray <- NULL
-  for (path in simulationResults$allQuantityPaths) {
-    endOfPath <- utils::tail(ospsuite::toPathArray(path), 3)
-    pathsArray <- rbind.data.frame(
-      pathsArray,
-      data.frame(
-        "parentCompartmentName" = endOfPath[1],
-        "subCompartmentName" = endOfPath[2],
-        "compoundName" = endOfPath[3],
-        "path" = path,
-        stringsAsFactors = FALSE
+  allAppliedCompoundNames <- simulation$allXenobioticFloatingMoleculeNames()
+  # Default configuration behavior integrates all xenobiotic molecules in mass balance
+  # Note that a list of lists is expected from the json file,
+  # thus the same format is used for default configuration
+  massBalanceSettings <- list(list(
+    Name = paste("Mass Balance for", paste(allAppliedCompoundNames, collapse = ", ")),
+    Molecules = allAppliedCompoundNames
+  ))
+  massBalanceSettings <- structureSet$simulationSet$massBalanceSettings %||% massBalanceSettings
+
+  massBalanceResults <- list()
+  for (plotSettings in massBalanceSettings) {
+    compoundNames <- as.character(plotSettings$Molecules)
+    # Sub section using settings name
+    if(!isEmpty(plotSettings$Name)){
+      sectionId <- defaultFileNames$resultID(length(massBalanceResults) + 1, "mass_balance")
+      massBalanceResults[[sectionId]] <- saveTaskResults(
+        id = sectionId,
+        textChunk = paste("###", plotSettings$Name, anchor(sectionId)),
+        includeTextChunk = TRUE
       )
+    }
```

Comment:
what happens if `plotSettings$Name` is empty? 
Somehow I cannot see the else-part (where the **default** subchapter caption is written)

Reply:
Currently, there is no sub section title added if `plotSettings$Name` is empty.